src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -0,0 +1,50 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.commands.DraftBasedCreateCommand;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+class BaseService<U, V> {
+    final BaseSyncOptions<U, V> syncOptions;
+    boolean isCached = false;
+    final Map<String, String> keyToIdCache = new ConcurrentHashMap<>();
```

Comment:
do we really expect to have access `keyToIdCache` in regular concurrent read/write mode, so it is implemented as `ConcurrentHashMap`?